### PR TITLE
PixelPaint: Fixes to LayerListWidget + Merge Layer Down action

### PR DESCRIPTION
**Pull request by mustafaquraish**
1. Fix the order in which layers are displayed, with the layers in front now appearing on top. 

2. Add a "Merge Layer Down" action, which combines a layer with the one below it.

3. Draw layers justified to the bottom instead of the top, and fix the spacing for the LayerListWidget a little to not clip the bottom gadget. Look below.

Before:
before

After:
after

### CODE REVIEW

File: Userland/Applications/PixelPaint/LayerListWidget.cpp [hunk 1]
```diff
@@ -226,22 +242,25 @@ void LayerListWidget::image_did_remove_layer(size_t layer_index)
...
+    auto gadget_index = m_image->layer_count() - layer_index;
+    m_gadgets.remove(gadget_index);
+    m_selected_gadget_index = to_gadget_index(0);
     relayout_gadgets();
 }
 
 void LayerListWidget::image_did_modify_layer_properties(size_t layer_index)
 {
-    update(m_gadgets[layer_index].rect);
+
```
Comment (AtkinsSJ) [inline Userland/Applications/PixelPaint/LayerListWidget.cpp:254]: Blank line here is a little weird but not a big deal

File: Userland/Applications/PixelPaint/main.cpp [hunk 2]
```diff
@@ -541,6 +541,19 @@ int main(int argc, char** argv)
         },
         window));
 
+    layer_menu.add_action(GUI::Action::create(
+        "M&erge Active Layer Down", { Mod_Ctrl, Key_E }, [&](auto&) {
```
Comment (AtkinsSJ) [inline Userland/Applications/PixelPaint/main.cpp:545]: Looks like Ctrl+E is copied from Photoshop? It doesn't make intuitive sense as a shortcut to me, but copying Photoshop is fair enough. :^)

Comment (mustafaquraish) [reply Userland/Applications/PixelPaint/main.cpp:545]: Yep, I'm not sure why it's Ctrl+E, but it's ingrained in my muscle memory so unless someone has a better idea I'm sticking to it

File: Userland/Applications/PixelPaint/LayerListWidget.cpp (hunk 1 above)
Comment (mustafaquraish) [reply Userland/Applications/PixelPaint/LayerListWidget.cpp]: fixed, thanks

**Pull request merged**